compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1502,11 +1891,46 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
 
                 let l = self.new_named_lifetime(lifetime.id, id, span, ident);
                 hir::GenericArg::Lifetime(l)
-            }));
+            })
+            .collect();
         debug!(?lifetimes);
 
+        let type_const_args = self.collect_type_const_args(&ast_generics.params);
+
+        let universal_args: Vec<_> = impl_trait_inputs.iter().map(|ty| match ty.kind {
+            TyKind::ImplTrait(node_id, _) => {
+                let def_id = self.local_def_id(node_id);
+                let span = ty.span;
+                let ident = Ident::from_str_and_span(&pprust::ty_to_string(ty), span);
+                hir::GenericArg::Type(hir::Ty {
+                    hir_id: self.next_id(),
+                    span,
```

Comment:
Related: this is currently missing a `lower_span`.

Reply:
https://github.com/rust-lang/rust/pull/101345/commits/835d7de9950804fa806aaf3ff6173b3299e99c43